### PR TITLE
Adding shaman attunement stuff

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Mods "Improving shaman attunement"

#### Purpose of change
Yknow, there is nothing here atm

#### Describe the solution

- [x] Add spells
- [x] fix the bug with mound, that avoid using the AoE spell attack
- [x] add `passive "mind reading" effect that shows monsters a certain distance away`

#### Testing
Works fine, find them pretty useful

### CODE REVIEW

**Comment by catdach** [comment]
> I genuinely have no clue why monster don't use `shambling_aoe`

It might need a to have cooldown?
Also, might need to be above level 0?
so like, try making it:
```
"special_attacks": [
      { "type": "spell", "spell_data": { "id": "shambling_aoe", "min_level": 1 }, cooldown: 25 },
```

**Comment by GuardianDll** [comment]
nope, both of leveling and cooldown do nothing, sadly

File: data/mods/Magiclysm/Spells/attunements/Shaman.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
+[
+  {
+    "id": "butterfly_swarm",
+    "type": "SPELL",
+    "name": { "str": "Butterfly Swarm" },
+    "description": "Summon pack of butterflies with razor-sharp wings, to cut every living creature on your way.",
```
Comment (schoolcf) [inline data/mods/Magiclysm/Spells/attunements/Shaman.json:6]: not sure why just living if its not bio
```suggestion
    "description": "Summon a pack of butterflies with razor-sharp wings, to cut every creature in your way.",
```

File: data/mods/Magiclysm/Spells/attunements/Shaman.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    "aoe_increment": 0.2,
+    "min_duration": 14400,
+    "max_duration": 144000,
+    "duration_increment": 4000
+  },
+  {
+    "id": "summon_shambling_mound",
+    "type": "SPELL",
+    "name": { "str": "Call of Mound" },
+    "description": "Summon the Shambling Mound, the giant carnivorous plant mess with devastating area attacks.  It is too slow for common fight, but it is the best defender for skilled druid.",
```
Comment (schoolcf) [inline data/mods/Magiclysm/Spells/attunements/Shaman.json:39]: droseras r sundews, basically a tentacle thing
```suggestion
    "description": "Summon a Shambling Mound, a giant Drosera with devastating area attacks.  It is very slow, but is an excellent defender.",
```

File: data/mods/Magiclysm/enchantments/Shaman.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "description": "You can predict some movements of your enemies (just slowdown in fact).  If you are seeing this spell, you have debugged it in.",
+    "valid_targets": [ "hostile" ],
+    "flags": [ "SILENT", "NO_PROJECTILE" ],
+    "effect": "attack",
+    "shape": "blast",
+    "effect_str": "mind_reading",
+    "min_damage": 0,
+    "max_damage": 0,
+    "min_aoe": 7,
+    "max_aoe": 7
```
Comment (schoolcf) [inline data/mods/Magiclysm/enchantments/Shaman.json:15]: i m very confsued pls help D:
"sense other minds" sounds like a "haha ik where u r" type thing and less a "i will read ur mind and dodge everything"
```suggestion
    "description": "You can predict some movements of your enemies, allowing you to easily dodge their attacks.  If you are seeing this spell, you have debugged it in.",
    "valid_targets": [ "self" ],
    "flags": [ "SILENT", "NO_PROJECTILE" ],
    "effect": "attack",
    "effect_str: "mind_reading",
    "shape": "blast",
    "min_damage": 0,
    "max_damage": 0,
    "min_range": "1",
    "max_range": "1",
    },
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 4]
```diff
@@ -539,5 +539,70 @@
     "death_function": { "corpse_type": "NO_CORPSE", "message": "The %s melts away." },
     "path_settings": { "avoid_traps": true, "avoid_sharp": true },
     "flags": [ "SEES", "WARM", "BASHES", "DESTROYS", "NO_BREATHE", "PUSH_MON", "PUSH_VEH", "FIREY", "PATH_AVOID_DANGER_1" ]
+  },
+  {
+    "id": "mon_razor_butterflies",
+    "type": "MONSTER",
+    "name": { "str": "razor butterfly", "str_pl": "razor butterflies" },
+    "//": "Now only for shaman, todo: make them natural spawn, add harvest, etc",
+    "description": "Some unknown species of butterfly, they have clear white wings with some orange rune pattern, and a razor-sharp part on the edge.  They are pretty big, but still in natural limits.",
```
Comment (schoolcf) [inline data/mods/Magiclysm/monsters/monsters.json:548]: i tried to do a dcss reference
idk if the dash should be there
```suggestion
    "description": "A large, brightly colored butterfly with crystalline wings, fluttering incongruously through the air.  This one seems to be patterned with orange runes, and its wings look razor-sharp.",
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 5]
```diff
@@ -539,5 +539,70 @@
...
+    "dodge": 8,
+    "luminance": 0,
+    "armor_bash": 0,
+    "armor_cut": 0,
+    "armor_bullet": 0,
+    "vision_day": 14,
+    "vision_night": 14,
+    "special_attacks": [ { "type": "leap", "cooldown": 3, "max_range": 2, "allow_no_target": true }, { "id": "cut_throat" } ],
+    "material": [ "iflesh" ],
+    "death_function": { "message": "A %s dissolve.", "corpse_type": "NO_CORPSE" },
```
Comment (schoolcf) [inline data/mods/Magiclysm/monsters/monsters.json:572]: s
```suggestion
    "death_function": { "message": "A %s dissolves.", "corpse_type": "NO_CORPSE" },
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 6]
```diff
@@ -539,5 +539,70 @@
...
+    "flags": [ "SEES", "HEARS", "NOHEAD", "HARDTOSHOOT", "FLIES", "NO_BREATHE", "NOGIB", "ATTACKMON", "STUN_IMMUNE", "SWARMS" ],
+    "//1": "has no HIT_AND_RUN cause they start to be too dumb"
+  },
+  {
+    "id": "mon_shambling_mound",
+    "type": "MONSTER",
+    "name": "shambling mound",
+    "description": "This giant plant mess is one of the ancients magic species in the world.  They don't like to walk too much, but the number of nutrients .",
```
Comment (schoolcf) [inline data/mods/Magiclysm/monsters/monsters.json:580]: add acid damage or smth idk
idk how to make it sound nice
```suggestion
    "description": "A magical, ancient Drosera.  A thick tangle of sap covered tentacles, though barely movable, it is capable of great destruction",
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 7]
```diff
@@ -539,5 +539,70 @@
...
+    "dodge": 0,
+    "luminance": 0,
+    "armor_bash": 20,
+    "armor_cut": 20,
+    "armor_bullet": 20,
+    "vision_day": 45,
+    "vision_night": 6,
+    "special_attacks": [ { "type": "spell", "spell_data": { "id": "shambling_aoe", "min_level": 0 }, "cooldown": 1 } ],
+    "material": [ "veggy" ],
+    "death_function": { "message": "A %s dissolve.", "corpse_type": "NO_CORPSE" },
```
Comment (schoolcf) [inline data/mods/Magiclysm/monsters/monsters.json:604]: sssss
```suggestion
    "death_function": { "message": "A %s dissolves.", "corpse_type": "NO_CORPSE" },
```

File: data/mods/Magiclysm/traits/attunements.json [hunk 8]
```diff
@@ -829,6 +829,8 @@
     "valid": false,
     "description": "Deeply attuned to the spiritual fabric that shapes and nurtures the natural world.  Shamans call upon the primeval spirits of nature for aid and advice.  Be careful however, for ancient spirits are fickle and unknowable things.",
```
Comment (schoolcf) [inline data/mods/Magiclysm/traits/attunements.json:830]: comma D:<
```suggestion
    "description": "Deeply attuned to the spiritual fabric that shapes and nurtures the natural world, shamans call upon the primeval spirits of nature for aid and advice.  Be careful however, for ancient spirits are fickle and unknowable things.",
```

File: data/mods/Magiclysm/Spells/attunements/Shaman.json (hunk 2 above)
Comment (GuardianDll) [reply data/mods/Magiclysm/Spells/attunements/Shaman.json:39]: It doesn't looks like drosera, just google some images, its from DnD

File: data/mods/Magiclysm/effects/effects.json [hunk 9]
```diff
@@ -311,6 +311,17 @@
...
+    "type": "effect_type",
+    "id": "mind_reading",
+    "//": "slowdown for shaman attunement",
+    "name": [ "" ],
+    "desc": [ "" ],
+    "max_intensity": 1,
+    "int_add_val": 1,
+    "max_duration": "6 s",
+    "base_mods": { "speed_mod": [ -25 ] }
+  },
```
[the remark this answers is not in the document]
Comment (GuardianDll) [reply data/mods/Magiclysm/effects/effects.json:324]: Good idea, but messages are 100% not supposed to be here, cause they would spam in chat every time you talk with person

File: data/mods/Magiclysm/monsters/monsters.json [hunk 10]
```diff
@@ -539,5 +539,70 @@
...
+    "material": [ "iflesh" ],
+    "death_function": { "message": "A %s dissolves.", "corpse_type": "NO_CORPSE" },
+    "flags": [ "SEES", "HEARS", "NOHEAD", "HARDTOSHOOT", "FLIES", "NO_BREATHE", "NOGIB", "ATTACKMON", "STUN_IMMUNE", "SWARMS" ],
+    "//1": "has no HIT_AND_RUN cause they start to be too dumb"
+  },
+  {
+    "id": "mon_shambling_mound",
+    "type": "MONSTER",
+    "name": "shambling mound",
+    "description": "This giant plant mess is one of the ancients magic species in the world.  They don't like to walk too much, but the number of nutrients .",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/monsters/monsters.json:580]: this description needs some work.

**Comment by KorGgenT** [comment]
fix up the descriptions and the rest of it's good.

File: data/mods/Magiclysm/monsters/monsters.json (hunk 10 above)
Comment (GuardianDll) [reply data/mods/Magiclysm/monsters/monsters.json]: Dunno how i missed that :derpy:

**Pull request merged**